api/src/domain/response.rs

```diff
@@ -0,0 +1,20 @@
+use serde::{Deserialize, Serialize};
+use sqlx;
+
+#[derive(Debug, sqlx::FromRow, sqlx::Encode, sqlx::Decode)]
+pub struct DBResponse {
+    pub id: String,
+    pub status_code: u16,
+    pub name: Option<String>,
+    #[sqlx(default)]
+    pub headers: Vec<ResponseHeader>,
+    pub body: Option<serde_json::Value>,
+}
+
+#[derive(
+    Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, Debug, sqlx::Encode, sqlx::Decode,
+)]
+pub struct ResponseHeader {
```

Comment:
Probably don't need this and can use a common struct for both request and response headers, but was lazy :)